<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Collect Results from Agamemnon</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">tcga_micro</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mleukam/tcga_micro">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Collect Results from Agamemnon</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>tcga_micro/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201014code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201014)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201014code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201014)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/Projects/tcga_micro
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamicrotreee74ce69ba752cf55a9839b020eb41321f0d6445ftargetblanke74ce69a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/tcga_micro/tree/e74ce69ba752cf55a9839b020eb41321f0d6445f" target="_blank">e74ce69</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamicrotreee74ce69ba752cf55a9839b020eb41321f0d6445ftargetblanke74ce69a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mleukam/tcga_micro/tree/e74ce69ba752cf55a9839b020eb41321f0d6445f" target="_blank">e74ce69</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    corrected_genus_cpm_expressionset.rds
    Ignored:    data/2021_01_11_readcounts.csv
    Ignored:    data/2021_01_20_readcounts.csv
    Ignored:    data/africa_metadata.csv
    Ignored:    data/africa_mode1.csv
    Ignored:    data/genus_analysis_list.rds
    Ignored:    data/imm_cor_souce.tsv
    Ignored:    data/mode2_tcga_counts_list.rds
    Ignored:    data/mode2_tcga_counts_list_clean.rds
    Ignored:    data/tcga_counts_list_mode2.rds
    Ignored:    data/tcga_proj_taxonomy.csv
    Ignored:    data/tcga_proj_taxonomy.rds
    Ignored:    data/tcga_proj_taxonomy_nomissing.csv
    Ignored:    data/tcga_proj_taxonomy_nomissing.rds
    Ignored:    data/totreads_list_backup.rds
    Ignored:    data/totreads_list_mode2.rds
    Ignored:    output/.DS_Store
    Ignored:    output/africa/
    Ignored:    output/display_genus_immune_correlations_pearson.xlsx
    Ignored:    output/display_genus_immune_correlations_spearman.xlsx
    Ignored:    output/display_genus_summary_correlations_spearman.xlsx
    Ignored:    output/draft_mode2_clindat.csv
    Ignored:    output/draft_mode2_counts_rowformat.csv
    Ignored:    output/genus_immune_correlations_pearson.rds
    Ignored:    output/genus_immune_correlations_spearman.rds
    Ignored:    output/genus_sumvar_correlations_pearson.rds
    Ignored:    output/genus_sumvar_correlations_spearman.rds
    Ignored:    output/global_high_level_correlations.csv
    Ignored:    output/mode2_counts_columnformat_draft.csv
    Ignored:    output/plots/
    Ignored:    output/tcga_batchcorrected_correlations_pearson.xlsx
    Ignored:    output/tcga_batchcorrected_correlations_spearman.xlsx
    Ignored:    output/tcga_correlations_pearson.xlsx
    Ignored:    output/tcga_correlations_spearman.xlsx
    Ignored:    output/tcga_global_batchcorrected_correlations_spearman.xlsx

Untracked files:
    Untracked:  analysis/africa_csf.Rmd
    Untracked:  code/get_reads_v2.sh

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/collect_tables.Rmd</code>) and HTML (<code>docs/collect_tables.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_micro/a4c88ae243732a44fee36e20c57c77a8e8fdeb1c/docs/collect_tables.html" target="_blank">a4c88ae</a>
</td>
<td>
Michael Leukam
</td>
<td>
2021-02-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_micro/blob/5ed0dcdc30fe7863bd344b789018e01d00d98ec5/analysis/collect_tables.Rmd" target="_blank">5ed0dcd</a>
</td>
<td>
Michael Leukam
</td>
<td>
2021-02-03
</td>
<td>
working code for scraping agamemnon results
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_micro/blob/4460b56404945565cb8b872324cb6700a6cb41e3/analysis/collect_tables.Rmd" target="_blank">4460b56</a>
</td>
<td>
Michael Leukam
</td>
<td>
2020-11-24
</td>
<td>
updates for mode2
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_micro/blob/bfcc57ff144d7b776faa4b2e235678958d25a13a/analysis/collect_tables.Rmd" target="_blank">bfcc57f</a>
</td>
<td>
Michael Leukam
</td>
<td>
2020-10-28
</td>
<td>
working correlation scripts for genus and summary variables
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_micro/blob/9bc98f0a56aa79f48267cc5c1163accb7afbd9e3/analysis/collect_tables.Rmd" target="_blank">9bc98f0</a>
</td>
<td>
Michael Leukam
</td>
<td>
2020-10-21
</td>
<td>
updates with cpm
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_micro/blob/ba97fe4710595798779b1793793c658c1692cce6/analysis/collect_tables.Rmd" target="_blank">ba97fe4</a>
</td>
<td>
Michael Leukam
</td>
<td>
2020-10-20
</td>
<td>
updated to collect results of get_reads.sh
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_micro/blob/9ea4930c6ad79634770abbaae9d0f879715cf61a/analysis/collect_tables.Rmd" target="_blank">9ea4930</a>
</td>
<td>
Michael Leukam
</td>
<td>
2020-10-20
</td>
<td>
updates to include kingdown and phylum data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_micro/blob/6162be468238d0fc50bd2cd17d50111a17ab99cb/analysis/collect_tables.Rmd" target="_blank">6162be4</a>
</td>
<td>
Michael Leukam
</td>
<td>
2020-10-14
</td>
<td>
initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Clear workspace On O2 cluster</p>
<pre class="r"><code>rm(list=ls())</code></pre>
<pre class="r"><code>knitr::opts_chunk$set(eval = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<p>Make repaired taxonomy</p>
<pre class="r"><code>setwd(&quot;~/Projects/tcga_micro&quot;)
# read in phylogenetic tree from agamemnon
taxonomy &lt;- read_csv(&quot;data/tcga_proj_taxonomy.csv&quot;)

# make repars to missing data in taxonomy tree
# if &quot;---&quot; is present, replace with reasonable substitute with flag
# eg Anelloviridae_Family
taxonomy_nomissing &lt;- taxonomy %&gt;%
  # all missing superkingdom data are viruses (manual check)
  # viruses are not classified in the same system as cellular life
  # will use term &#39;virus&#39; for all viruses at superkingdom level
  mutate(Superkingdom = ifelse(Superkingdom == &quot;---&quot;, &quot;Virus&quot;, Superkingdom)) %&gt;%
  # if genus is missing, fill in with a higher level
  mutate(Genus = 
    ifelse((Genus == &quot;---&quot; &amp; Family != &quot;---&quot;), paste0(Family, &quot;_Family&quot;),
      ifelse((Genus == &quot;---&quot; &amp; Order != &quot;---&quot;), paste0(Order, &quot;_Order&quot;),
        ifelse((Genus == &quot;---&quot; &amp; Class != &quot;---&quot;), paste0(Class, &quot;_Class&quot;),
          ifelse((Genus == &quot;---&quot; &amp; Phylum != &quot;---&quot;), paste0(Phylum, &quot;_Phylum&quot;),
            ifelse((Genus == &quot;---&quot; &amp; Superkingdom != &quot;---&quot;), paste0(Superkingdom, &quot;_Superkingdom&quot;),
            Genus)))))) %&gt;%
  # if family is missing, fill in with a higher level
  mutate(Family = 
    ifelse((Family == &quot;---&quot; &amp; Order != &quot;---&quot;), paste0(Order, &quot;_Order&quot;),
      ifelse((Family == &quot;---&quot; &amp; Class != &quot;---&quot;), paste0(Class, &quot;_Class&quot;),
        ifelse((Family == &quot;---&quot; &amp; Phylum != &quot;---&quot;), paste0(Phylum, &quot;_Phylum&quot;),
          ifelse((Family == &quot;---&quot; &amp; Superkingdom != &quot;---&quot;), paste0(Superkingdom, &quot;_Superkingdom&quot;),
          Family))))) %&gt;%
  # if order is missing, fill in with a higher level
  mutate(Order = 
      ifelse((Order == &quot;---&quot; &amp; Class != &quot;---&quot;), paste0(Class, &quot;_Class&quot;),
        ifelse((Order == &quot;---&quot; &amp; Phylum != &quot;---&quot;), paste0(Phylum, &quot;_Phylum&quot;),
          ifelse((Order == &quot;---&quot; &amp; Superkingdom != &quot;---&quot;), paste0(Superkingdom, &quot;_Superkingdom&quot;),
          Order)))) %&gt;%
  # if class is missing, fill in with a higher level
  mutate(Class = 
      ifelse((Class == &quot;---&quot; &amp; Phylum != &quot;---&quot;), paste0(Phylum, &quot;_Phylum&quot;),
        ifelse((Class == &quot;---&quot; &amp; Superkingdom != &quot;---&quot;), paste0(Superkingdom, &quot;_Superkingdom&quot;),
          Class))) %&gt;%
  # if phylum is missing, fill in with a higher level
  mutate(Phylum = 
      ifelse((Phylum == &quot;---&quot; &amp; Superkingdom != &quot;---&quot;), paste0(Superkingdom, &quot;_Superkingdom&quot;),
          Phylum)) 

# save copy
taxonomy_nomissing %&gt;% saveRDS(&quot;data/tcga_proj_taxonomy_nomissing.rds&quot;)
taxonomy_nomissing %&gt;% write_csv(&quot;data/tcga_proj_taxonomy_nomissing.csv&quot;)
# move taxonomy nomissing to HPC with filezilla
# location: ~/tcga_micro in my personal storage</code></pre>
<p>Setup on O2 cluster</p>
<pre class="bash"><code>srun --pty -p interactive --mem 64G -t 0-06:00 /bin/bash
module load gcc/6.2.0
module load R/4.0.1
cd /n/scratch3/users/m/mw292/AGAMEMNON/TCGA/Results2
R</code></pre>
</div>
<div id="define-function" class="section level2">
<h2>Define function</h2>
<pre class="r"><code># test variables
input_dir &lt;- &quot;/n/scratch3/users/m/mw292/AGAMEMNON/TCGA/Results2/TCGA_HNSC&quot;
output_dir &lt;- &quot;/n/scratch3/users/m/mjl62/agamemnon_counts/mode2&quot;
tcga_project &lt;- &quot;HNSC&quot;</code></pre>
<pre class="r"><code>gather_counts &lt;- function(input_dir, tcga_project){
  #
  # function to gather all agamemnon output files
  # path to directories as a quoted string as dir arguments
  # ok if nested in subdirectories
  # 4-letter TCGA project as project argument
  # before starting, ensure no duplicate file_ids
  # output is three new CSVs:
  #   counts table
  #   clinical data
  #   taxonomy key
  #
  # load required packages
  library(tidyverse)
  library(GenomicDataCommons)
  library(Matrix.utils)
  #
  # set constants: 
  output_dir = &quot;/n/scratch3/users/m/mjl62/agamemnon_counts/mode2&quot;
  #
  # read in repaired taxonomy with no missing data
  #
  # print starting message
  print(paste0(&quot;starting analysis of TCGA project &quot;, tcga_project))
  #
  # get dataframe of paths to results.tab files with file uuids
  paths_df &lt;- dir(path = input_dir, 
           recursive = TRUE, 
           include.dirs = FALSE, 
           full.names = TRUE) %&gt;%
    enframe() %&gt;%
    dplyr::filter(grepl(&quot;results.tab&quot;, value)) %&gt;%
    separate(value, into = c(NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, &quot;file_id&quot;),
             sep = &quot;/&quot;, remove = FALSE) %&gt;%
    dplyr::select(pathto = value, file_id)
  # get clinical data from file uuids
  # requires GenomicDataCommons
  file_uuids_vector &lt;- paths_df %&gt;%
    pull(file_id)
  conversion_table &lt;- files(legacy = FALSE) %&gt;%
    GenomicDataCommons::filter( ~ file_id %in% file_uuids_vector) %&gt;%
    GenomicDataCommons::select(c(&quot;cases.case_id&quot;,&quot;cases.samples.submitter_id&quot;)) %&gt;%
    results_all() %&gt;%
    as_tibble() %&gt;%
    unnest(cases) %&gt;%
    unnest(samples) %&gt;%
    dplyr::select(submitter_id, file_id, case_id) %&gt;%
    separate(submitter_id, into = c(&quot;x1&quot;, &quot;x2&quot;, &quot;x3&quot;, NA), sep = &quot;-&quot;) %&gt;%
    unite(tcga_barcode, &quot;x1&quot;, &quot;x2&quot;, &quot;x3&quot;, sep = &quot;-&quot;)
  clin_data &lt;- conversion_table %&gt;%
    pull(case_id) %&gt;%
    gdc_clinical() %&gt;%
    reduce(left_join, by = &quot;case_id&quot;)
  # get sample type out of GDC
  sample_type &lt;- files(legacy = FALSE) %&gt;%
    GenomicDataCommons::filter( ~ file_id %in% file_uuids_vector) %&gt;%
    GenomicDataCommons::select(&quot;cases.samples.sample_type&quot;) %&gt;%
    results_all() %&gt;%
    as_tibble() %&gt;%
    unnest(cases) %&gt;%
    unnest(samples)
  # get aliquot ID out of GDC
  aliquot_id &lt;- files(legacy = FALSE) %&gt;% 
    GenomicDataCommons::filter( ~ file_id %in% file_uuids_vector) %&gt;% 
    GenomicDataCommons::select(&quot;cases.samples.portions.analytes.aliquots.submitter_id&quot;) %&gt;%
    results_all() %&gt;%
    as_tibble() %&gt;%
    unnest(cases) %&gt;%
    unnest(samples) %&gt;%
    unnest(portions) %&gt;%
    unnest(analytes) %&gt;%
    unnest(aliquots) %&gt;%
    dplyr::select(file_id, aliquot_submitter_id = submitter_id)
  clin_df &lt;- list(conversion_table, clin_data) %&gt;%
    reduce(left_join, by = &quot;case_id&quot;) %&gt;%
    left_join(sample_type, by = &quot;file_id&quot;) %&gt;%
    left_join(aliquot_id, by = &quot;file_id&quot;) %&gt;%
    dplyr::select(file_id, 
                  tcga_barcode,
                  tcga_full_barcode = aliquot_submitter_id,
                  case_id, 
                  sample_type, 
                  everything())
  # make lookup data for colnames
  barcode_lookuptable &lt;- clin_df %&gt;%
    dplyr::select(tcga_barcode, file_id, case_id)
  #
  # read tsvs to a list of tibbles
  paths_list &lt;- paths_df %&gt;%
    pull(pathto) %&gt;%
    as.list()
  names(paths_list) &lt;- paths_df$file_id
  tsv_list &lt;- map(paths_list, read_tsv)
  # pull off taxonomy labels and set aside
  taxonomy &lt;- tsv_list[[1]] %&gt;%
    dplyr::select(Superkingdom, 
                  Phylum, 
                  Class, 
                  Order, 
                  Family, 
                  Genus, 
                  Species, 
                  Scientific_Name, 
                  TaxID) %&gt;%
    mutate(TaxID = as.character(TaxID))
  # read in taxonomy
  taxonomy_nomissing &lt;- read_csv(&quot;~/tcga_micro/tcga_proj_taxonomy_nomissing.csv&quot;) %&gt;%
    # expects all variables to be characters for downstream merging
    mutate(TaxID = as.character(TaxID))
  # get matrix of counts with file_id as colnames
  tsv_mat &lt;- tsv_list %&gt;%
    map(function(df){df %&gt;% pull(NumCounts)}) %&gt;%
    bind_cols() %&gt;%
    as.matrix()
  rownames(tsv_mat) &lt;- taxonomy$TaxID
  #
  # aggregate by rownames using Matrix.utils
  # add column for TCGA barcodes
  # add column for sample_type
  taxid_counts_tbl &lt;- aggregate.Matrix(tsv_mat, row.names(tsv_mat)) %&gt;%
    as.matrix() %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(var = &quot;TaxID&quot;) %&gt;%
    as_tibble()
  # counts by species
  species_counts_intermed &lt;- taxid_counts_tbl %&gt;%
    left_join(taxonomy_nomissing, by = &quot;TaxID&quot;) %&gt;%
    dplyr::select(Species, everything()) %&gt;%
    dplyr::select(-Superkingdom, 
                  -Phylum, 
                  -Class,
                  -Order, 
                  -Family,
                  -Genus,
                  -Scientific_Name,
                  -TaxID)
  species_counts_mat &lt;- species_counts_intermed %&gt;%
    dplyr::select(-Species) %&gt;%
    as.matrix()
  rownames(species_counts_mat) &lt;- species_counts_intermed %&gt;%
    pull(Species)
  species_counts_tbl &lt;- aggregate.Matrix(species_counts_mat, row.names(species_counts_mat)) %&gt;%
    as.matrix() %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(var = &quot;Species&quot;) %&gt;%
    as_tibble()
  # aggregate by rownames using Matrix.utils
  # add column for TCGA barcodes
  # add column for sample_type
  # counts by genus
  genus_counts_intermed &lt;- taxid_counts_tbl %&gt;%
    # join taxonomy lookup table
    left_join(taxonomy_nomissing, by = &quot;TaxID&quot;) %&gt;%
    # remove undesired columns
    dplyr::select(-Superkingdom, 
                  -Phylum, 
                  -Class,
                  -Order, 
                  -Family,
                  -Species,
                  -Scientific_Name,
                  -TaxID)
  # make a matrix with count labels as rownames
  genus_counts_mat &lt;- genus_counts_intermed %&gt;%
    dplyr::select(-Genus) %&gt;%
    as.matrix()
  rownames(genus_counts_mat) &lt;- genus_counts_intermed %&gt;%
    pull(Genus)
  # aggregate by summation for matching rownames
  genus_counts_tbl &lt;- aggregate.Matrix(genus_counts_mat, row.names(genus_counts_mat)) %&gt;%
    as.matrix() %&gt;%
    # convert back to dataframe
    as.data.frame() %&gt;%
    rownames_to_column(var = &quot;Genus&quot;) %&gt;%
    as_tibble()
  # counts by kingdom
  king_counts_intermed &lt;- taxid_counts_tbl %&gt;%
    left_join(taxonomy_nomissing, by = &quot;TaxID&quot;) %&gt;%
    dplyr::select(Species, everything()) %&gt;%
    dplyr::select(-Phylum, 
                  -Class,
                  -Order, 
                  -Family,
                  -Genus,
                  -Species,
                  -Scientific_Name,
                  -TaxID)
  king_counts_mat &lt;- king_counts_intermed %&gt;%
    dplyr::select(-Superkingdom) %&gt;%
    as.matrix()
  rownames(king_counts_mat) &lt;- king_counts_intermed %&gt;%
    pull(Superkingdom)
  king_counts_tbl &lt;- aggregate.Matrix(king_counts_mat, row.names(king_counts_mat)) %&gt;%
    as.matrix() %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(var = &quot;Superkingdom&quot;) %&gt;%
    as_tibble()
   # counts by phylum
  phylum_counts_intermed &lt;- taxid_counts_tbl %&gt;%
    left_join(taxonomy_nomissing, by = &quot;TaxID&quot;) %&gt;%
    dplyr::select(Species, everything()) %&gt;%
    dplyr::select(-Superkingdom, 
                  -Class,
                  -Order, 
                  -Family,
                  -Genus,
                  -Species,
                  -Scientific_Name,
                  -TaxID)
  phylum_counts_mat &lt;- phylum_counts_intermed %&gt;%
    dplyr::select(-Phylum) %&gt;%
    as.matrix()
  rownames(phylum_counts_mat) &lt;- phylum_counts_intermed %&gt;%
    pull(Phylum)
  phylum_counts_tbl &lt;- aggregate.Matrix(phylum_counts_mat, row.names(phylum_counts_mat)) %&gt;%
    as.matrix() %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(var = &quot;Phylum&quot;) %&gt;%
    as_tibble()
  # counts by taxid/scientific_name
  taxid_counts_intermed &lt;- taxid_counts_tbl %&gt;%
    left_join(taxonomy_nomissing, by = &quot;TaxID&quot;) %&gt;%
    dplyr::select(Scientific_Name, everything()) %&gt;%
    dplyr::select(-Superkingdom, 
                  -Class,
                  -Order, 
                  -Family,
                  -Genus,
                  -Species,
                  -Phylum,
                  -TaxID)
  taxid_counts_mat &lt;- taxid_counts_intermed %&gt;%
    dplyr::select(-Scientific_Name) %&gt;%
    as.matrix()
  rownames(taxid_counts_mat) &lt;- taxid_counts_intermed %&gt;%
    pull(Scientific_Name)
  taxid_counts_tbl &lt;- aggregate.Matrix(taxid_counts_mat, row.names(taxid_counts_mat)) %&gt;%
    as.matrix() %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(var = &quot;Scientific Name&quot;) %&gt;%
    as_tibble()
  #
  #
  # write taxID counts out as CSV
  write_csv(taxid_counts_tbl, 
            paste0(output_dir, 
                   &quot;/&quot;, 
                   tcga_project, 
                   &quot;_taxID_counts.csv&quot;))
  # write Species counts out as CSV
  write_csv(species_counts_tbl, 
            paste0(output_dir, 
                   &quot;/&quot;, 
                   tcga_project, 
                   &quot;_species_counts.csv&quot;))
  # write Genus counts out as CSV
  write_csv(genus_counts_tbl, 
            paste0(output_dir, 
                   &quot;/&quot;, 
                   tcga_project, 
                   &quot;_genus_counts.csv&quot;))
  write_csv(king_counts_tbl,
            paste0(output_dir,
                   &quot;/&quot;,
                   tcga_project,
                   &quot;_kingdom_counts.csv&quot;))
  write_csv(phylum_counts_tbl,
            paste0(output_dir,
                   &quot;/&quot;,
                   tcga_project,
                   &quot;_phylum_counts.csv&quot;))
  # write clinical data out as CSV
  write_csv(clin_df,  
            paste0(output_dir, 
                   &quot;/&quot;, 
                   tcga_project, 
                   &quot;_clindata.csv&quot;))
}</code></pre>
</div>
<div id="apply-function" class="section level2">
<h2>Apply function</h2>
<p>On O2 (interactive)</p>
<pre class="r"><code># navigate to target directory
setwd(&quot;/n/scratch3/users/m/mw292/AGAMEMNON/TCGA/Results2&quot;)

# make table of input variables
ftab_nostad &lt;- list.files() %&gt;%
  as_tibble() %&gt;%
  dplyr::rename(directory = value) %&gt;%
  dplyr::filter(grepl(&quot;TCGA_&quot;, directory)) %&gt;%
  separate(directory, into = c(NA, &quot;tcga_project&quot;, NA), remove = FALSE) %&gt;%
  mutate(input_dir = paste0(&quot;/n/scratch3/users/m/mw292/AGAMEMNON/TCGA/Results2/&quot;, 
                            directory)) %&gt;%
  dplyr::select(input_dir, tcga_project) %&gt;%
  # remove projects with missing data
  dplyr::filter(tcga_project != &quot;STAD&quot;) %&gt;%
  dplyr::filter(tcga_project != &quot;PRAD&quot;) %&gt;%
  print() 

# add stad project labeled &quot;mode2&quot;
ftab_stad &lt;- list.files() %&gt;%
  as_tibble() %&gt;%
  dplyr::rename(directory = value) %&gt;%
  dplyr::filter(grepl(&quot;TCGA_&quot;, directory)) %&gt;%
  separate(directory, into = c(NA, &quot;tcga_project&quot;, NA), remove = FALSE) %&gt;%
  mutate(input_dir = paste0(&quot;/n/scratch3/users/m/mw292/AGAMEMNON/TCGA/Results2/&quot;, 
                            directory)) %&gt;%
  dplyr::select(input_dir, tcga_project) %&gt;%
  # remove projects with missing data
  dplyr::filter(grepl(&quot;_mode2&quot;, input_dir)) %&gt;%
  print() 

# combine tables 
ftab &lt;- ftab_nostad %&gt;%
  bind_rows(ftab_stad) %&gt;%
  print()

# apply gather function to table
map2(ftab$input_dir, ftab$tcga_project, gather_counts)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
