---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

Analysis of TCGA data using Agamemnon

## Data prep

Read in data from Agememnon output. Supplement with clinical data from GDC and 
immune correlates table provided by Nelly. Taxonomy data repaired by filling missing taxa with the lowest available taxonomic class above the missing level. Summary indices and high-level taxa counts (e.g. viral vs bacterial) calculated for each case. Cases filtered for primary tumors only and only one biopsy per case ID was kept. Raw counts from Agamemnon mode 2 summarized at genus level (or if genus is not available, at the lowest available taxonomic group at genus or above). Cases with no counts removed. Converted to counts per million total original (human and microbial) reads (extracted from Agamemnon logs) Missing counts converted to zeros. TMM normalization factors used for voom normalization with tumor type set as experimental design. Counts normalized by voom batch corrected using limma::removeBatchEffect using tumor type as experimental design and combination of sequencing center and plate (extracted from TCGA barcode) as batch effect. Constant added to all counts to eliminate negative values after batch correction.

* [Collect Tables](collect_tables.html) - get individual results from Agamemnon into a single table for eeach project (run on O2). Repair taxonomy with no missing taxa.
* [TCGA Data Prep](tcga_data_prep.html) - change rownames to TCGA legacy barcodes. Get full TCGA barcodes for batch correction. Add in clinical and immune correlate data. Save output in list structure. Filter cases for primary tumors only and eliminate second biopsies from cases with multiple tissue biopsies.
* [Derive Variables](derive_vars.html) - calculate measures of diversity, richness, high level taxonomy, and other basic descriptive summary variables for later analysis.
* [Batch Correction](batch_correction.html) - log transformation, normalization, and batch correction by sequencing center and plate using limma model.

## Immune correlates

* [Calculate immune correlations for genera](genus_models.html)

## Survival analysis


