---
title: "tcga_data_prep"
author: "Michael Leukam"
date: "2020-09-12"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Setup

Clear the workspace and set seed
```{r}
rm(list=ls())
set.seed(818)
```

Load packages
```{r}
# data cleaning
library(tidyverse)
library(Matrix.utils)
```

```{r}
knitr::opts_chunk$set(eval = FALSE)
```

Setup on local
```{bash, eval = FALSE}
sshfs mjl62@o2.hms.harvard.edu:/n/scratch3/users/m/mjl62 ~/sshfs_mount/TCGA

# get immune correlates
cd ~/sshfs_mount/TCGA/agamemnon_counts

curl https://contattafiles.s3.us-west-1.amazonaws.com/tnt40761/jOjcXzknipvEC1q/Characteristics.tsv > IMM_correlates.tsv
```

Load packages
```{r}
# data cleaning
library(tidyverse)
```

## COUNTS
```{r}
# READ IN--------------------------------------------
# make list of TCGA projects
tcga_proj <- list("STAD" = "STAD",
                  "LUAD" = "LUAD",
                  "BRCA" = "BRCA",
                  "OV" = "OV",
                  "ESCA" = "ESCA",
                  "KIRK" = "KIRC",
                  "HNSC" = "HNSC")
# second batch
tcga_proj <- list("ACC" = "ACC",
                  "BLCA" = "BLCA",
                  "CESC" = "CESC",
                  "COAD" = "COAD",
                  "KICH" = "KICH",
                  "KIRP" = "KIRP",
                  "LAML" = "LAML",
                  "LIHC" = "LIHC",
                  "MESO" = "MESO",
                  "PAAD" = "PAAD",
                  "PCPG" = "PCPG",
                  "PRAD" = "PRAD",
                  "READ" = "READ",
                  "SARC" = "SARC",
                  "SKCM" = "SKCM",
                  "STAD" = "STAD",
                  "TGCT" = "TGCT",
                  "THCA" = "THCA",
                  "THYM" = "THYM",
                  "UCS" = "UCS",
                  "UVM" = "UVM")


# function to read in data for each project
file_loader <- function(tcga_proj){
  # assign path to data files
  path_to_directory <- "~/sshfs_mount/TCGA/agamemnon_counts/"
  # read in genus counts
  genus_counts <- read_csv(
  paste0(path_to_directory,
         tcga_proj, "_genus_counts.csv")
  )
   # read in species counts
  species_counts <- read_csv(
  paste0(path_to_directory, 
         tcga_proj, "_species_counts.csv")
  )
   # read in taxid counts
  taxid_counts <- read_csv(
    paste0(path_to_directory,
           tcga_proj, "_taxID_counts.csv")
  )
  # read in phylum counts
  phylum_counts <- read_csv(
    paste0(path_to_directory,
           tcga_proj, "_phylum_counts.csv")
  )
  # read in kingdom counts
  kingdom_counts <- read_csv(
    paste0(path_to_directory,
           tcga_proj, "_kingdom_counts.csv")
  )
  # read in clinical data
  clindat <- read_csv(
    paste0(path_to_directory, 
           tcga_proj, "_clindata.csv")
  ) %>%
    # keep only primary tumor data
    dplyr::filter(sample_type == "Primary Tumor") %>%
    # convert explicit attributes table to standard tbl
    # honestly because it is annoying for str() calls
    # and does not look pretty
    select(everything())
  # make counts tables into a list for batch processing
  counts <- list(genus_counts, 
                 species_counts, 
                 taxid_counts, 
                 kingdom_counts, 
                 phylum_counts)
  names(counts) <- c("genus", "species", "taxid", "kingdom", "phylum")
  # return a list of counts (nested) and clinical data
  combined_list <- list("counts" = counts, "clindat" = clindat)
  combined_list
}
# apply function to list
counts_clin_list <- purrr::map(tcga_proj, file_loader)
str(counts_clin_list, max.level = 2)

# FIX ROWNAMES--------------------------------------------
# function to convert file_ids to legacy TCGA barcodes
# using conversion columns in clindata
convert_to_barcodes <- function(list_as_element){
  clindat <- list_as_element[["clindat"]]
  imm_cor <- list_as_element[["imm_cor"]]
  
  lookup_table <- clindat %>% dplyr::select(file_id, 
                                            tcga_barcode, 
                                            sample_type)
  counts_tables <- list_as_element[["counts"]]
  counts_barcodes <- map(counts_tables, function(df){
    mat <- df[-1] %>%
      as.matrix()
    rownames(mat) <- df %>% dplyr::pull(var = 1)
    mat %>%
      t() %>%
      as.data.frame() %>%
      tibble::rownames_to_column(var = "file_id") %>%
      tibble::as_tibble() %>%
      dplyr::left_join(lookup_table) %>%
      dplyr::select(file_id, tcga_barcode, sample_type, everything()) %>%
      # select only primary tumors for now
      dplyr::filter(sample_type == "Primary Tumor") %>%
      dplyr::select(-file_id, -sample_type)
  })
  list("genus" = counts_barcodes[["genus"]], 
       "species" = counts_barcodes[["species"]],
       "taxid" = counts_barcodes[["taxid"]],
       "kingdom" = counts_barcodes[["kingdom"]],
       "phylum" = counts_barcodes[["phylum"]],
       "clindat" = clindat)
}
# apply function to list
tcga_barcodes_list <- purrr::map(counts_clin_list, convert_to_barcodes)
str(tcga_barcodes_list, max.level = 2)

# IMMUNE CORRELATES--------------------------------------------
# read in
path_to_directory <- "~/sshfs_mount/TCGA/agamemnon_counts/"
imm_corr <- read_tsv(
  paste0(path_to_directory, "IMM_correlates.tsv")
  ) %>% 
  dplyr::select(tcga_barcode = `TCGA Participant Barcode`, everything())

# make immune correlates list
imm_corr_lister <- function(tcga_proj){
  imm_corr %>% dplyr::filter(`TCGA Study` == tcga_proj)
}
imm_corr_list <- purrr::map(tcga_proj, imm_corr_lister)

# LISTS BY TYPE--------------------------------------------
# functions to split lists by count types
genera_lister <- function(list_element){
  genera <- list_element[["genus"]]
genera
}
species_lister <- function(list_element){
  species <- list_element[["species"]]
  species
}
taxid_lister <- function(list_element){
  taxid <- list_element[["taxid"]]
  taxid
}
kingdom_lister <- function(list_element){
  kingdom <- list_element[["kingdom"]]
  kingdom
}
phylum_lister <- function(list_element){
  phylum <- list_element[["phylum"]]
  phylum
}
# apply functions to list
genera_list <- purrr::map(tcga_barcodes_list, genera_lister)
species_list <- purrr::map(tcga_barcodes_list, species_lister)
taxid_list <- purrr::map(tcga_barcodes_list, taxid_lister)
phylum_list <- purrr::map(tcga_barcodes_list, phylum_lister)
kingdom_list <- purrr::map(tcga_barcodes_list, kingdom_lister)

# create nested list of counts
counts_sublist <- list("kingdom" = kingdom_list,
                       "phylum" = phylum_list,
                       "genus" = genera_list,
                       "species" = species_list, 
                       "taxid" = taxid_list)

# add in clinical data and immune correlates
clindat_lister <- function(list_element){
  clindat <- list_element[["clindat"]]
  clindat
}
clindat_list <- purrr::map(tcga_barcodes_list, clindat_lister)

tcga_counts <- list("counts" = counts_sublist,
                    "clindat" = clindat_list,
                    "imm_cor" = imm_corr_list)

# SAVE FOR LATER USE---------------------------------------
saveRDS(tcga_counts, "data/tcga_counts_list.rds")

```

Check integrity 
```{r}
tcga_counts <- readRDS("data/tcga_counts_list.rds")
str(tcga_counts$counts, max.level = 2)
str(tcga_counts, max.level = 2)
tcga_counts$counts$genus$ACC
tcga_counts$counts$species$ACC
tcga_counts$counts$kingdom$ACC
tcga_counts$counts$taxid$ACC
```

## TOTAL READS (LIBRARY SIZE)

```{r}
reads_loader <- function(tcga_proj){
  # assign path to data files
  path_to_directory <- "~/sshfs_mount/TCGA/agamemnon_counts/"
  # read in genus counts
  genus_counts <- read_csv(
    paste0(path_to_directory, 
           tcga_proj, 
           "_totalreads.csv")
  )}

tcga_reads <- list("ACC"  = "ACC",
                  "BLCA" = "BLCA",
                  "CESC" = "CESC",
                  "ESCA" = "ESCA",
                  "GBM"  =  "GBM",
                  "HNSC" = "HNSC",
                  "KICH" = "KICH",
                  "KIRP" = "KIRP",
                  "LAML" = "LAML",
                  "LGG"  = "LGG",
                  "LUAD" = "LUAD",
                  "LIHC" = "LIHC",
                  "OV"   = "OV",
                  "PCPG" = "PCPG",
                  "PRAD" = "PRAD",
                  "READ" = "READ",
                  "SARC" = "SARC",
                  "SKCM" = "SKCM",
                  "STAD" = "STAD",
                  "TGCT" = "TGCT",
                  "THCA" = "THCA",
                  "THYM" = "THYM",
                  "UCS" = "UCS",
                  "UVM" = "UVM")

totreads_list <- map(tcga_reads, reads_loader)
str(totreads_list, max.level = 2)

# save as backup
saveRDS(totreads_list, "data/totreads_list_backup.rds")
```

## Combine first and second batch
```{r}
batch_one <- readRDS("~/Projects/ag_microbiome/data/tcga_counts_list.rds")
batch_two <- readRDS("~/Projects/tcga_micro/data/tcga_counts_list.rds")

dropped <- which(names(batch_two$counts$genus) %in% c("STAD", "LAML"))
dropped
comb_genus <- c(batch_one$counts$genus, batch_two$counts$genus[-dropped])
str(comb_genus, max.level = 2)

comb_species <- c(batch_one$counts$species, batch_two$counts$species[-dropped])
str(comb_species, max.level = 2)

comb_taxid <- c(batch_one$counts$species, batch_two$counts$taxid[-dropped])
str(comb_taxid, max.level = 2)

comb_kingdom <- c(batch_one$counts$kingdom, batch_two$counts$kingdom[-dropped])
str(comb_kingdom, max.level = 2)

comb_phylum <- c(batch_one$counts$phylum, batch_two$counts$phylum[-dropped])
str(comb_phylum, max.level = 2)

comb_counts <- list("genus" = comb_genus,
                 "species" = comb_species,
                 "taxid" = comb_taxid,
                 "kingdom" = comb_kingdom,
                 "phylum" = comb_phylum)
str(comb_counts, max.level = 2)

comb_clindat <- c(batch_one$clindat, batch_two$clindat[-dropped])
str(comb_clindat, max.level = 2)

comb_imm_cor <- c(batch_one$imm_cor, batch_two$imm_cor[-dropped])
str(comb_imm_cor, max.level = 2)

comb_tcga_counts <- list("counts" = comb_counts, 
                         "clindat" = comb_clindat, 
                         "imm_cor" = comb_imm_cor)
str(comb_tcga_counts, max.level = 2)
```

## Repair missing kingdom/phylum data
read in phylogenetic tree from agamemnon
```{r}
taxonomy <- read_tsv("~/sshfs_mount/TCGA/agamemnon_counts/results_example.tab") %>%
  dplyr::select(-NumCounts)

# Make list of inputs
stad <- comb_tcga_counts$counts$taxid$STAD
coad <- comb_tcga_counts$counts$taxid$COAD
esca <- comb_tcga_counts$counts$taxid$ESCA
hnsc <- comb_tcga_counts$counts$taxid$HNSC
laml <- comb_tcga_counts$counts$taxid$LAML
luad <- comb_tcga_counts$counts$taxid$LUAD
ov <- comb_tcga_counts$counts$taxid$OV

taxid_list <- list(
  "STAD" = stad,
  "ESCA" = esca,
  "HNSC" = hnsc,
  "OV" = ov
)

king_count_repairer <- function(taxid_list_element){
  taxonomy <- read_tsv(
    "~/sshfs_mount/TCGA/agamemnon_counts/results_example.tab") %>%
    dplyr::select(-NumCounts)
  king_counts_intermed <- taxid_list_element %>%
    as.data.frame() %>%
    column_to_rownames(var = "tcga_barcode") %>%
    as.matrix() %>%
    t() %>%
    as.data.frame() %>%
    rownames_to_column(var = "Scientific_Name") %>%
    left_join(taxonomy, by = "Scientific_Name") %>%
    dplyr::select(-Phylum, 
                  -Class,
                  -Order, 
                  -Family,
                  -Genus,
                  -Species,
                  -Scientific_Name,
                  -TaxID) %>%
    dplyr::select(Superkingdom, everything())
  king_counts_mat <- king_counts_intermed %>%
    dplyr::select(-Superkingdom) %>%
    as.matrix()
  rownames(king_counts_mat) <- king_counts_intermed %>%
    pull(Superkingdom)
  king_counts_tbl <- aggregate.Matrix(king_counts_mat,
                                      row.names(king_counts_mat)) %>%
    as.matrix() %>%
    t() %>%
    as.data.frame() %>%
    rownames_to_column(var = "tcga_barcode") %>%
    as_tibble()
  king_counts_tbl
}
# apply function
king_repair_list <- map(taxid_list, king_count_repairer)

# phylum repair
phylum_count_repairer <- function(taxid_list_element){
  taxonomy <- read_tsv(
    "~/sshfs_mount/TCGA/agamemnon_counts/results_example.tab") %>%
    dplyr::select(-NumCounts)
  phylum_counts_intermed <- taxid_list_element %>%
    as.data.frame() %>%
    column_to_rownames(var = "tcga_barcode") %>%
    as.matrix() %>%
    t() %>%
    as.data.frame() %>%
    rownames_to_column(var = "Scientific_Name") %>%
    left_join(taxonomy, by = "Scientific_Name") %>%
    dplyr::select(-Superkingdom, 
                  -Class,
                  -Order, 
                  -Family,
                  -Genus,
                  -Species,
                  -Scientific_Name,
                  -TaxID) %>%
    dplyr::select(Phylum, everything())
  phylum_counts_mat <- phylum_counts_intermed %>%
    dplyr::select(-Phylum) %>%
    as.matrix()
  rownames(phylum_counts_mat) <- phylum_counts_intermed %>%
    pull(Phylum)
  phylum_counts_tbl <- aggregate.Matrix(phylum_counts_mat,
                                      row.names(phylum_counts_mat)) %>%
    as.matrix() %>%
    t() %>%
    as.data.frame() %>%
    rownames_to_column(var = "tcga_barcode") %>%
    as_tibble()
  phylum_counts_tbl
}
# apply function
phylum_repair_list <- map(taxid_list, phylum_count_repairer)

# add in to master list
comb_tcga_counts$counts$kingdom$STAD <- king_repair_list$STAD
comb_tcga_counts$counts$kingdom$ESCA <- king_repair_list$ESCA
comb_tcga_counts$counts$kingdom$HNSC <- king_repair_list$HNSC
comb_tcga_counts$counts$kingdom$OV <- king_repair_list$OV

comb_tcga_counts$counts$phylum$STAD <- phylum_repair_list$STAD
comb_tcga_counts$counts$phylum$ESCA <- phylum_repair_list$ESCA
comb_tcga_counts$counts$phylum$HNSC <- phylum_repair_list$HNSC
comb_tcga_counts$counts$phylum$OV <- phylum_repair_list$OV
```

## ADD IN LIBRARY SIZE
https://rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/#:~:text=Count%20up%20the%20total%20reads,of%20the%20gene%2C%20in%20kilobases.
```{r}
totreads_list<- readRDS("data/totreads_list_backup.rds")

totreads_tbl <- totreads_list %>%
  enframe() %>%
  unnest(value) %>%
  dplyr::select(file_id, reads) %>%
  print()
clindat_tbl_list <- comb_tcga_counts$clindat %>%
  enframe %>%
  unnest(value) %>%
  left_join(totreads_tbl, by = "file_id") %>%
  dplyr::select(name, tcga_barcode, totreads = reads, everything()) %>%
  group_by(name) %>%
  nest() %>%
  deframe()
str(clindat_tbl, max.level = 1)

# convert counts to CPM
reads_lookup <- clindat_tbl_list %>%
  enframe() %>%
  unnest(value) %>%
  dplyr::select(tcga_barcode, totreads) %>%
  print()

# genus
genusflat <- comb_tcga_counts$counts$genus %>%
  enframe() %>%
  unnest(value) %>%
  left_join(reads_lookup) %>%
  dplyr::select(name, tcga_barcode, totreads, everything()) %>%
  # take out those lacking total reads data
  dplyr::filter(!is.na(totreads)) %>%
  print()
genus_cpm <- genusflat %>%
  # get per million scaling factor
  mutate(factor = totreads / 10^6) %>%
  select(-totreads) %>%
  # divide by readcounts
  mutate(across(-c(factor, name, tcga_barcode), ~ . / factor)) %>%
  dplyr::select(-factor) %>%
  group_by(name) %>%
  nest() %>%
  deframe()

# species
speciesflat <- comb_tcga_counts$counts$species %>%
  enframe() %>%
  unnest(value) %>%
  left_join(reads_lookup) %>%
  dplyr::select(name, tcga_barcode, totreads, everything()) %>%
  # take out those lacking total reads data
  dplyr::filter(!is.na(totreads)) %>%
  print()
species_cpm <- speciesflat %>%
  # get per million scaling factor
  mutate(factor = totreads / 10^6) %>%
  select(-totreads) %>%
  # divide by readcounts
  mutate(across(-c(factor, name, tcga_barcode), ~ . / factor)) %>%
  dplyr::select(-factor) %>%
  group_by(name) %>%
  nest() %>%
  deframe()

# phylum
phylumflat <- comb_tcga_counts$counts$phylum %>%
  enframe() %>%
  unnest(value) %>%
  left_join(reads_lookup) %>%
  dplyr::select(name, tcga_barcode, totreads, everything()) %>%
  # take out those lacking total reads data
  dplyr::filter(!is.na(totreads)) %>%
  print()
phylum_cpm <- phylumflat %>%
  # get per million scaling factor
  mutate(factor = totreads / 10^6) %>%
  select(-totreads) %>%
  # divide by readcounts
  mutate(across(-c(factor, name, tcga_barcode), ~ . / factor)) %>%
  dplyr::select(-factor) %>%
  group_by(name) %>%
  nest() %>%
  deframe()

# taxid
taxidflat <- comb_tcga_counts$counts$taxid %>%
  enframe() %>%
  unnest(value) %>%
  left_join(reads_lookup) %>%
  dplyr::select(name, tcga_barcode, totreads, everything()) %>%
  # take out those lacking total reads data
  dplyr::filter(!is.na(totreads)) %>%
  print()
taxid_cpm <- taxidflat %>%
  # get per million scaling factor
  mutate(factor = totreads / 10^6) %>%
  select(-totreads) %>%
  # divide by scaling factor
  mutate(across(-c(factor, name, tcga_barcode), ~ . / factor)) %>%
  dplyr::select(-factor) %>%
  group_by(name) %>%
  nest() %>%
  deframe()
taxid_cpm

kingdomflat <- comb_tcga_counts$counts$kingdom %>%
  enframe() %>%
  unnest(value) %>%
  left_join(reads_lookup) %>%
  dplyr::select(name, tcga_barcode, totreads, everything()) %>%
  dplyr::select(-`NA`) %>%
  # take out those lacking total reads data
  dplyr::filter(!is.na(totreads)) %>%
  print()
kingdom_cpm <- kingdomflat %>%
  # get per million scaling factor
  mutate(factor = totreads / 10^6) %>%
  select(-totreads) %>%
  # divide by scaling factor
  mutate(across(-c(factor, name, tcga_barcode), ~ . / factor)) %>%
  dplyr::select(-factor) %>%
  group_by(name) %>%
  nest() %>%
  deframe()
kingdom_cpm

cpm_counts <- list("kingdom_cpm" = kingdom_cpm, 
                   "phylum_cpm" = phylum_cpm, 
                   "genus_cpm" = genus_cpm, 
                   "species_cpm" = species_cpm, 
                   "taxid_cpm" = taxid_cpm)

comb_tcga_counts[["cpm_counts"]] <- cpm_counts

str(comb_tcga_counts, max.level = 2)

saveRDS(comb_tcga_counts, "data/comb_tcga_counts_list.rds")
```
