---
title: "tcga_data_prep"
author: "Michael Leukam"
date: "2021-02-02"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Setup

Clear the workspace and set seed
```{r}
rm(list=ls())
set.seed(818)
```

```{r}
knitr::opts_chunk$set(eval = FALSE)
```

Load packages
```{r}
# data cleaning
library(tidyverse)
library(Biobase)
library(Matrix.utils)
library(limma)
library(ggpubr)
library(edgeR)
```

Setup on local
```{bash, eval = FALSE}
sshfs mjl62@o2.hms.harvard.edu:/n/scratch3/users/m/mjl62 ~/sshfs_mount/TCGA

# get immune correlates
cd ~/sshfs_mount/TCGA/agamemnon_counts

curl https://contattafiles.s3.us-west-1.amazonaws.com/tnt40761/jOjcXzknipvEC1q/Characteristics.tsv > IMM_correlates.tsv
```

## Format counts
```{r}
# READ IN--------------------------------------------
setwd("~/sshfs_mount/TCGA/agamemnon_counts/mode2")

# make list of TCGA projects
tcga_proj <- list("BRCA" = "BRCA",
                  "LUAD" = "LUAD",
                  "GBM" = "GBM",
                  "LUSC" = "LUSC",
                  "SKCM" = "SKCM",
                  "STAD" = "STAD",
                  "LGG" = "LGG",
                  "UCEC" = "UCEC",
                  "KIRC" = "KIRC",
                  # "LAML" = "LAML", no samples in this folder
                  "HNSC" = "HNSC",
                  "THCA" = "THCA")
# second batch
# tcga_proj <- list("ACC" = "ACC",
#                  "BLCA" = "BLCA",
#                  "CESC" = "CESC",
#                  "COAD" = "COAD",
#                  "KICH" = "KICH",
#                  "KIRP" = "KIRP",
#                  "LAML" = "LAML",
#                  "LIHC" = "LIHC",
#                  "MESO" = "MESO",
#                  "PAAD" = "PAAD",
#                  "PCPG" = "PCPG",
#                  "PRAD" = "PRAD",
#                  "READ" = "READ",
#                  "SARC" = "SARC",
#                  "SKCM" = "SKCM",
#                  "STAD" = "STAD",
#                  "TGCT" = "TGCT",
#                  "THCA" = "THCA",
#                  "THYM" = "THYM",
#                  "UCS" = "UCS",
#                  "UVM" = "UVM")


# function to read in data for each project
file_loader <- function(tcga_proj){
  # assign path to data files
  path_to_directory <- "~/sshfs_mount/TCGA/agamemnon_counts/mode2/"
  # read in genus counts
  genus_counts <- read_csv(
  paste0(path_to_directory,
         tcga_proj, "_genus_counts.csv")
  )
   # read in species counts
  species_counts <- read_csv(
  paste0(path_to_directory, 
         tcga_proj, "_species_counts.csv")
  )
   # read in taxid counts
  taxid_counts <- read_csv(
    paste0(path_to_directory,
           tcga_proj, "_taxID_counts.csv")
  )
  # read in phylum counts
  phylum_counts <- read_csv(
    paste0(path_to_directory,
           tcga_proj, "_phylum_counts.csv")
  )
  # read in kingdom counts
  kingdom_counts <- read_csv(
    paste0(path_to_directory,
           tcga_proj, "_kingdom_counts.csv")
  )
  # read in clinical data
  clindat <- read_csv(
    paste0(path_to_directory, 
           tcga_proj, "_clindata.csv")
  ) %>%
    # keep only primary tumor data
    dplyr::filter(sample_type == "Primary Tumor") %>%
    # convert explicit attributes table to standard tbl
    # honestly because it is annoying for str() calls
    # and does not look pretty
    select(everything())
  # make counts tables into a list for batch processing
  counts <- list(genus_counts, 
                 species_counts, 
                 taxid_counts, 
                 kingdom_counts, 
                 phylum_counts)
  names(counts) <- c("genus", "species", "taxid", "kingdom", "phylum")
  # return a list of counts (nested) and clinical data
  combined_list <- list("counts" = counts, "clindat" = clindat)
  combined_list
}
# apply function to list
counts_clin_list <- purrr::map(tcga_proj, file_loader)
str(counts_clin_list, max.level = 2)

# FIX ROWNAMES--------------------------------------------
# function to convert file_ids to legacy TCGA barcodes
# using conversion columns in clindata
convert_to_barcodes <- function(list_as_element){
  clindat <- list_as_element[["clindat"]]
  imm_cor <- list_as_element[["imm_cor"]]
  
  lookup_table <- clindat %>% dplyr::select(file_id, 
                                            tcga_barcode, 
                                            sample_type)
  counts_tables <- list_as_element[["counts"]]
  counts_barcodes <- map(counts_tables, function(df){
    mat <- df[-1] %>%
      as.matrix()
    rownames(mat) <- df %>% dplyr::pull(var = 1)
    mat %>%
      t() %>%
      as.data.frame() %>%
      tibble::rownames_to_column(var = "file_id") %>%
      tibble::as_tibble() %>%
      dplyr::left_join(lookup_table) %>%
      dplyr::select(file_id, tcga_barcode, sample_type, everything()) %>%
      # select only primary tumors for now
      dplyr::filter(sample_type == "Primary Tumor") %>%
      dplyr::select(-file_id, -sample_type)
  })
  list("genus" = counts_barcodes[["genus"]], 
       "species" = counts_barcodes[["species"]],
       "taxid" = counts_barcodes[["taxid"]],
       "kingdom" = counts_barcodes[["kingdom"]],
       "phylum" = counts_barcodes[["phylum"]],
       "clindat" = clindat)
}
# apply function to list
tcga_barcodes_list <- purrr::map(counts_clin_list, convert_to_barcodes)
str(tcga_barcodes_list, max.level = 2)

# IMMUNE CORRELATES--------------------------------------------
# read in locally saved copy
imm_corr <- read_tsv("~/Projects/tcga_micro/data/imm_cor_souce.tsv") %>%
  dplyr::select(tcga_barcode = `TCGA Participant Barcode`, everything())

# make immune correlates list
imm_corr_lister <- function(tcga_proj){
  imm_corr %>% dplyr::filter(`TCGA Study` == tcga_proj)
}
imm_corr_list <- purrr::map(tcga_proj, imm_corr_lister)

# LISTS BY TYPE--------------------------------------------
# functions to split lists by count types
genera_lister <- function(list_element){
  genera <- list_element[["genus"]]
genera
}
species_lister <- function(list_element){
  species <- list_element[["species"]]
  species
}
taxid_lister <- function(list_element){
  taxid <- list_element[["taxid"]]
  taxid
}
kingdom_lister <- function(list_element){
  kingdom <- list_element[["kingdom"]]
  kingdom
}
phylum_lister <- function(list_element){
  phylum <- list_element[["phylum"]]
  phylum
}
# apply functions to list
genera_list <- purrr::map(tcga_barcodes_list, genera_lister)
species_list <- purrr::map(tcga_barcodes_list, species_lister)
taxid_list <- purrr::map(tcga_barcodes_list, taxid_lister)
phylum_list <- purrr::map(tcga_barcodes_list, phylum_lister)
kingdom_list <- purrr::map(tcga_barcodes_list, kingdom_lister)

# create nested list of counts
counts_sublist <- list("kingdom" = kingdom_list,
                       "phylum" = phylum_list,
                       "genus" = genera_list,
                       "species" = species_list, 
                       "taxid" = taxid_list)

# add in clinical data and immune correlates
clindat_lister <- function(list_element){
  clindat <- list_element[["clindat"]]
  clindat
}
clindat_list <- purrr::map(tcga_barcodes_list, clindat_lister)

tcga_counts <- list("counts" = counts_sublist,
                    "clindat" = clindat_list,
                    "imm_cor" = imm_corr_list)

# SAVE FOR LATER USE---------------------------------------
saveRDS(tcga_counts, "~/Projects/tcga_micro/data/tcga_counts_list_mode2.rds")

```

Check integrity 
```{r}
tcga_counts <- readRDS("~/Projects/tcga_micro/data/tcga_counts_list_mode2.rds")
str(tcga_counts$counts, max.level = 2)
str(tcga_counts, max.level = 2)
tcga_counts$counts$genus$STAD
# problem - Tissierella_Class is still called ---_Class
tcga_counts$counts$genus <- tcga_counts$counts$genus %>% 
  map(~ dplyr::rename(., Tissierella_Class = `---_Class`))
tcga_counts$counts$genus$STAD
tcga_counts$counts$species$STAD
tcga_counts$counts$kingdom$STAD
tcga_counts$counts$taxid$STAD

```

## Data Cleaning

#### Convert to CPM
https://rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/#:~:text=Count%20up%20the%20total%20reads,of%20the%20gene%2C%20in%20kilobases.
```{r}
setwd("~/Projects/tcga_micro")
# read in readcounts from /code/get_reads_v2.sh
# moved over from HPC to data folder using filezilla
totreads_tbl <- read_csv("data/2021_01_20_readcounts.csv", col_names = FALSE) %>%
  dplyr::rename(tcga_proj = X1,
                file_id = X2,
                totreads = X3) %>%
  # STAD results were stored in folder named "STAD_mode2"
  # need to fix project names
  # remove leftover old STAD values
  dplyr::filter(!tcga_proj == "TCGA_STAD") %>%
  # remove `TCGA_` prefix from projects and _mode2 suffix
  separate(tcga_proj, into = c(NA, "tcga_proj", NA), sep = "_", ) %>%
  print()
  
summary(as.factor(totreads_tbl$tcga_proj))

# add total reads to clinical data using file_id
clindat_tbl_list <- tcga_counts$clindat %>%
  enframe %>%
  unnest(value) %>%
  left_join(totreads_tbl, by = "file_id") %>%
  dplyr::select(name, tcga_barcode, totreads, everything()) %>%
  group_by(name) %>%
  nest() %>%
  deframe()
str(clindat_tbl_list, max.level = 1)
clindat_tbl_list[1]

# make simplified lookup table
reads_lookup <- clindat_tbl_list %>%
  enframe() %>%
  unnest(value) %>%
  dplyr::select(tcga_barcode, totreads) %>%
  print()

# genus CPM conversion
genusflat <- tcga_counts$counts$genus %>%
  enframe() %>%
  unnest(value) %>%
  left_join(reads_lookup) %>%
  dplyr::select(name, tcga_barcode, totreads, everything()) %>%
  # take out those lacking total reads data
  dplyr::filter(!is.na(totreads)) %>%
  print()
genus_cpm <- genusflat %>%
  # get per million scaling factor
  mutate(factor = totreads / 10^6) %>%
  select(-totreads) %>%
  # divide by readcounts
  mutate(across(-c(factor, name, tcga_barcode), ~ . / factor)) %>%
  dplyr::select(-factor) %>%
  group_by(name) %>%
  nest() %>%
  deframe()

# species
speciesflat <- tcga_counts$counts$species %>%
  enframe() %>%
  unnest(value) %>%
  left_join(reads_lookup) %>%
  dplyr::select(name, tcga_barcode, totreads, everything()) %>%
  # take out those lacking total reads data
  dplyr::filter(!is.na(totreads)) %>%
  print()
species_cpm <- speciesflat %>%
  # get per million scaling factor
  mutate(factor = totreads / 10^6) %>%
  select(-totreads) %>%
  # divide by readcounts
  mutate(across(-c(factor, name, tcga_barcode), ~ . / factor)) %>%
  dplyr::select(-factor) %>%
  group_by(name) %>%
  nest() %>%
  deframe()

# phylum
phylumflat <- tcga_counts$counts$phylum %>%
  enframe() %>%
  unnest(value) %>%
  left_join(reads_lookup) %>%
  dplyr::select(name, tcga_barcode, totreads, everything()) %>%
  # take out those lacking total reads data
  dplyr::filter(!is.na(totreads)) %>%
  print()
phylum_cpm <- phylumflat %>%
  # get per million scaling factor
  mutate(factor = totreads / 10^6) %>%
  select(-totreads) %>%
  # divide by readcounts
  mutate(across(-c(factor, name, tcga_barcode), ~ . / factor)) %>%
  dplyr::select(-factor) %>%
  group_by(name) %>%
  nest() %>%
  deframe()

# taxid
taxidflat <- tcga_counts$counts$taxid %>%
  enframe() %>%
  unnest(value) %>%
  left_join(reads_lookup) %>%
  dplyr::select(name, tcga_barcode, totreads, everything()) %>%
  # take out those lacking total reads data
  dplyr::filter(!is.na(totreads)) %>%
  print()
taxid_cpm <- taxidflat %>%
  # get per million scaling factor
  mutate(factor = totreads / 10^6) %>%
  select(-totreads) %>%
  # divide by scaling factor
  mutate(across(-c(factor, name, tcga_barcode), ~ . / factor)) %>%
  dplyr::select(-factor) %>%
  group_by(name) %>%
  nest() %>%
  deframe()
taxid_cpm

kingdomflat <- tcga_counts$counts$kingdom %>%
  enframe() %>%
  unnest(value) %>%
  left_join(reads_lookup) %>%
  dplyr::select(name, tcga_barcode, totreads, everything()) %>%
  # dplyr::select(-`NA`) %>%
  # take out those lacking total reads data
  dplyr::filter(!is.na(totreads)) %>%
  print()
kingdom_cpm <- kingdomflat %>%
  # get per million scaling factor
  mutate(factor = totreads / 10^6) %>%
  select(-totreads) %>%
  # divide by scaling factor
  mutate(across(-c(factor, name, tcga_barcode), ~ . / factor)) %>%
  dplyr::select(-factor) %>%
  group_by(name) %>%
  nest() %>%
  deframe()
kingdom_cpm

cpm_counts <- list("kingdom_cpm" = kingdom_cpm, 
                   "phylum_cpm" = phylum_cpm, 
                   "genus_cpm" = genus_cpm, 
                   "species_cpm" = species_cpm, 
                   "taxid_cpm" = taxid_cpm,
                   "cpm_clindat" = clindat_tbl_list)

tcga_counts[["cpm_counts"]] <- cpm_counts

str(tcga_counts, max.level = 2)

saveRDS(tcga_counts, "data/mode2_tcga_counts_list_clean.rds")
```


## ALT START Batch Correction

From the manual for limma::removeBatchEffect:

This function is useful for removing batch effects, associated with hybridization time or other technical variables, prior to clustering or unsupervised analysis such as PCA, MDS or heatmaps. It is not intended to use with linear modelling. For linear modelling, it is better to include the batch factors in the linear model.

The design matrix is used to describe comparisons between the samples, for example treatment effects, which should not be removed.

The function (in effect) fits a linear model to the data, including both batches and regular treatments, then removes the component due to the batch effects.

The data object x can be of any class for which lmFit works. If x contains weights, then these will be used in estimating the batch effects.

```{r}
tcga_counts <- readRDS("data/mode2_tcga_counts_list_clean.rds")
str(tcga_counts, max.level = 2)
str(tcga_counts$cpm_counts, max.level = 2)

# pull out CPM counts
cpm_list <- tcga_counts$cpm_counts
str(cpm_list, max.level = 2)

# get linked clinical data to set up limma model
# convert to dataframe with subjects in rows
cpm_clindat <- tcga_counts$cpm_counts$cpm_clindat %>%
  # collapse list to single tbl
  enframe() %>%
  unnest(value) %>%
  # remove duplicates (very few)
  distinct(tcga_barcode, .keep_all = TRUE) %>%
  # create variable for plate and center == batch
  separate(tcga_full_barcode, 
           into = c(NA, NA, NA, NA, NA, "plate", "center"), 
           sep = "-", 
           remove = FALSE) %>%
  unite("batch", 
        c("plate", "center"), 
        sep = "", 
        remove = FALSE) %>%
  # convert to df with rownames
  as.data.frame() %>%
  column_to_rownames(var = "tcga_barcode") %>%
  print()
```

#### Genus
```{r}
# start with genus as code test
# get counts to set up limma model
genus_cpm <- cpm_list$genus_cpm %>%
  # collapse list to single tbl
  bind_rows() %>%
  # remove duplicates (very few, about 30)
  distinct(tcga_barcode, .keep_all = TRUE) %>%
  # convert to matrix and rotate
  # cases are in columns and features are in rows
  as.data.frame() %>%
  column_to_rownames(var = "tcga_barcode") %>%
  as.matrix() %>%
  t() 

# check results
dim(genus_cpm)
genus_cpm[1:5, 1:5]

# repair clinical data to match genus cpm
clindat_genus_tbl <- cpm_clindat %>%
  rownames_to_column(var = "tcga_barcode") %>%
  as_tibble()
clindat_genus_cpm <- enframe(colnames(genus_cpm)) %>%
  dplyr::select(tcga_barcode = value) %>%
  left_join(clindat_genus_tbl) %>%
  as.data.frame() %>%
  column_to_rownames(var = "tcga_barcode") %>%
  print()

# ensure that rows and columns are matched
identical(colnames(genus_cpm), rownames(clindat_genus_cpm))
length(rownames(clindat_genus_cpm))
length(colnames(genus_cpm))
```

Plots - can skip
```{r}
# ensure that data approximates normal distribution for limma
# make tidy dataframe for plotting
tidy_generator <- function(listelement){
  listelement %>% 
    pivot_longer(-tcga_barcode, names_to = "genus", values_to = "counts") %>%
    arrange(desc(counts))
}
# make tidy frames with cpm
tidy_genera <- map(cpm_list$genus_cpm, tidy_generator)

# counts distribution
counts_dist_plotter <- function(df){
  plot <- df %>%
    ggplot(aes(x = counts, color = tcga_barcode)) +
      geom_density() +
      theme(legend.position = "none")
  
}
# plot raw distributions of cpm for each tumor type
dist_plot_list <- map(tidy_genera, counts_dist_plotter)
do.call(ggarrange, dist_plot_list) %>%
  annotate_figure(., top = text_grob("Counts Distribution by Genus", color = "black", face = "bold", size = 14))
          
# log cpm distribution
log_dist_plotter <- function(df){
  plot <- df %>%
    mutate(lcounts = log2(counts)) %>%
    dplyr::filter(lcounts > -100) %>%
    ggplot(aes(x = lcounts), color = tcga_barcode) +
      geom_density() +
      theme(legend.position = "none") +
      expand_limits(x = c(-20, 30))
}
# log cpm density plots
log_plot_list <- map(tidy_genera, log_dist_plotter)
names(log_plot_list) <- names(tidy_genera)
do.call(ggarrange, log_plot_list) %>%
  annotate_figure(., top = text_grob("log(Counts) Distribution by Genus", color = "black", face = "bold", size = 14))

# log counts have a more normal distribution. unlogged counts have right skew
```

Limma batch correction
```{r}
#--------------------------------
## VOOM NORMALIZATION AND BATCH CORRECTION WITH LIMMA ##

# should pass non-log counts to voom
# voom includes a log2 transformation
# convert missing counts to 0
summary(as.factor(is.na(genus_cpm)))
summary(as.factor(is.nan(genus_cpm)))
genus_cpm[is.na(genus_cpm)] <- 0
summary(as.factor(is.na(genus_cpm)))
summary(as.factor(is.nan(genus_cpm)))
summary(as.factor(is.finite(genus_cpm)))
genus_cpm_finite <- genus_cpm[, genus_cpm[genus_cpm == Inf]]

# eliminate samples with library size 0
dim(genus_cpm)
genus_cpm_filt <- genus_cpm[, colSums(genus_cpm != 0) > 0]
dim(genus_cpm_filt)

# repair clinical data to match genus cpm after cases were dropped
clindat_genus_tbl <- cpm_clindat %>%
  rownames_to_column(var = "tcga_barcode") %>%
  as_tibble()
clindat_genus_cpm_filt <- enframe(colnames(genus_cpm_filt)) %>%
  dplyr::select(tcga_barcode = value) %>%
  left_join(clindat_genus_tbl) %>%
  as.data.frame() %>%
  column_to_rownames(var = "tcga_barcode") %>%
  print()

dim(clindat_genus_cpm_filt)

# ensure that rows and columns are matched
identical(colnames(genus_cpm_filt), rownames(clindat_genus_cpm_filt))
length(rownames(clindat_genus_cpm_filt))
length(colnames(genus_cpm_filt))

# create DGEList
dge <- DGEList(counts = genus_cpm_filt)

# apply TMM normalization to DGE
dge <- calcNormFactors(dge)

# set up design matrix
# design matrix should include real differences to be preserved
# Rob Knight paper used tumor type as real difference
# I will do the same
rnames <- rownames(clindat_genus_cpm_filt)
TCGA_ <- clindat_genus_cpm_filt$tcga_proj %>% as.factor()
summary(TCGA_)
design <- model.matrix(~ 0 + TCGA_)
rownames(design) <- rnames
head(design)

# Include sequencing center / plate as a batch variable to the model
batch <- as.factor(clindat_genus_cpm_filt$batch)
summary(batch)

# Remove batch effects
v <- voom(dge, design, plot = TRUE)
genus_cpm_corrected <- removeBatchEffect(v, batch = batch, design = design)

# check for negative results
genus_cpm_corrected[genus_cpm_corrected < 0] %>% length()
# what is lowest negative value?
genus_cpm_corrected %>% min()
# add correction factor to eliminate negative values
genus_cpm_corrected_noneg <- as.matrix(genus_cpm_corrected) + 31
genus_cpm_corrected_noneg %>% min()
# check for missing results
summary(as.factor(is.finite(genus_cpm_corrected_noneg)))

# save results as ExpressionSet
# first need variable annotation
v_metadata <- data.frame(
  labelDescription = colnames(clindat_genus_cpm_filt),
  row.names = colnames(clindat_genus_cpm_filt))

# make annotated phenoData
phenoData <- new("AnnotatedDataFrame", 
                 data = clindat_genus_cpm_filt, 
                 varMetadata = v_metadata)
phenoData

# convert expression data to matrix format
genus_cpm_mat <- genus_cpm_corrected_noneg %>% as.matrix()

# create global annotation
annotation <- as.character("Raw counts from Agamemnon mode 2 summarized at genus level (or if genus is not available, at the lowest available taxonomic group at genus or above). Cases with no counts removed. Duplicate cases removed. Converted to counts per million total original (human and microbial) reads (extracted from Agamemnon logs) Missing counts converted to zeros. TMM normalization factors used for voom normalization with tumor type set as experimental design. Counts normalized by voom batch corrected using limma::removeBatchEffect using tumor type as experimental design and combination of sequencing center and plate (extracted from TCGA barcode) as batch effect. Constant of 31 added to all counts to eliminate negative values after batch correction")

# assemble expressionset
genus_es <- ExpressionSet(
  assayData = genus_cpm_mat,
  phenoData = phenoData,
  annotation = annotation)
genus_es

# save
saveRDS(genus_es, "corrected_genus_cpm_expressionset.rds")
```
